framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileParser.java

```diff
@@ -1526,6 +1578,161 @@ private void putNewElement(
+     */
+    boolean sameType(TypeMirror javacType, Type javaParserType) {
+
+        switch (javacType.getKind()) {
+            case BOOLEAN:
+                return javaParserType.equals(PrimitiveType.booleanType());
+            case BYTE:
+                return javaParserType.equals(PrimitiveType.byteType());
+            case CHAR:
+                return javaParserType.equals(PrimitiveType.charType());
+            case DOUBLE:
+                return javaParserType.equals(PrimitiveType.doubleType());
+            case FLOAT:
+                return javaParserType.equals(PrimitiveType.floatType());
+            case INT:
+                return javaParserType.equals(PrimitiveType.intType());
+            case LONG:
+                return javaParserType.equals(PrimitiveType.longType());
+            case SHORT:
+                return javaParserType.equals(PrimitiveType.shortType());
+
+            case DECLARED:
+            case TYPEVAR:
+                if (!(javaParserType instanceof ClassOrInterfaceType)) {
+                    return false;
+                }
+                com.sun.tools.javac.code.Type javacTypeInternal =
+                        (com.sun.tools.javac.code.Type) javacType;
+                ClassOrInterfaceType javaParserClassType = (ClassOrInterfaceType) javaParserType;
+
+                // Use asString() because toString() includes annotations.
+                String javaParserString = javaParserClassType.asString();
+                Element javacElement = javacTypeInternal.asElement();
+                // Check both fully-qualified name and simple name.
+                return javacElement.toString().equals(javaParserString)
+                        || javacElement.getSimpleName().contentEquals(javaParserString);
+
+            case ARRAY:
+                return javaParserType.isArrayType()
+                        && sameType(
+                                ((ArrayType) javacType).getComponentType(),
+                                javaParserType.asArrayType().getComponentType());
+
+            default:
+                throw new BugInCF("Unhandled type %s of kind %s", javacType, javacType.getKind());
+        }
+    }
+
+    /**
+     * Process a fake override: copy its annotations to the fake overrides part of {@code
+     * #annotationFileAnnos}.
+     *
+     * @param element a real element
+     * @param decl a fake override of the element
+     * @param fakeLocation where the fake override was defined
+     */
+    private void processFakeOverride(
+            ExecutableElement element, CallableDeclaration<?> decl, TypeElement fakeLocation) {
+        // This is a fresh type, which this code may side-effect.
+        AnnotatedExecutableType methodType = atypeFactory.getAnnotatedType(element);
+
+        // TODO: Walk the type and the declaration, copying annotations from the declaration to the
+        // element.  I think PR #3799 has a visitor that does that, which I should use afte it is
```

Comment:
I don't think you mean https://github.com/typetools/checker-framework/pull/3799 here.